samples/python/llm/utils/model_utils.py

```diff
@@ -0,0 +1,145 @@
+        ov_config['CACHE_DIR'] = ""
+    # OpenVINO self have default value 2 for nstreams on machine with 2 nodes. Reducing memory consumed via changing nstreams to 1.
+    if 'NUM_STREAMS' not in ov_config:
+        ov_config['NUM_STREAMS'] = "1"
+
+def analyze_args(args):
+    model_args = {}
+    model_args['prompt'] = args.text if (args.text != None and args.text.endswith(".jsonl") == False) else args.prompt
+    model_args['prompt_file'] = args.text if (args.text != None and args.text.endswith(".jsonl")) else args.prompt_file
+    model_args["model_id"] = args.model_id
+    model_args['infer_count'] = args.infer_count
+    model_args["images"] = args.images
+    model_args["seed"] = args.seed
+    model_args["mem_consumption"] = args.memory_consumption
+    model_args["batch_size"] = args.batch_size
+    model_args['fuse_decoding_strategy'] = args.fuse_decoding_strategy
+    model_args['make_stateful'] = args.make_stateful
+    model_args['save_prepared_model'] = args.save_prepared_model
+    model_args['num_beams'] = args.num_beams
+    model_args['fuse_cache_reorder'] = args.fuse_cache_reorder
+
+    model_path = args.model
+    model_framework = args.framework
+    model_path = Path(model_path)
+    if not model_path.exists():
+        raise RuntimeError(f"==Failure FOUND==: Incorrect model path:{model_path}")
+    if model_framework == "ov":
+        model_names = args.model.split("/")
+        use_case, model_name = get_use_case(model_names)
+    elif model_framework == "pt":
+        if len(model_args['model_id']) != 0:
+            model_args['model_id'] = args.model_id
+        model_names = args.model.split("/")
+        use_case, model_name = get_use_case(model_names)
+    model_args['use_case'] = use_case
+    if use_case == "code_gen" and not model_args["prompt"] and not model_args["prompt_file"]:
+        model_args["prompt"] = "def print_hello_world():"
+    model_args["config"] = {}
+    if args.load_config != None:
+        config = get_config(args.load_config)
+        if type(config) is dict and len(config) > 0:
+            model_args["config"] = config
+    if model_framework == "ov":
+        set_default_param_for_ov_config(model_args["config"])
+        print(f"ov_config={model_args['config']}")
+    elif model_framework == "pt":
+        print(f"pt_config={model_args['config']}")
+    model_args["model_type"] = get_model_type(model_name, use_case, model_framework)
+    model_args["model_name"] = model_name
+    return model_path, model_framework, model_args, model_name
+
+def get_use_case(model_name_list):
+    for model_name in model_name_list:
+        for case, model_ids in USE_CASES.items():
+            for model_id in model_ids:
+                if model_id in model_name:
+                    print("==SUCCESS FOUND==: use_case: {}, model_name: {}".format(case, model_name))
+                    return case, model_name
+    raise RuntimeError("==Failure FOUND==: no use_case found")
```

Comment:
Do we need to remove the highlight symbol "=="? I am not sure.

Reply:
I am not sure, will discuss this issue.